src/components/pages/HomePage.tsx

```diff
@@ -22,14 +24,34 @@ const HomePage = (): JSX.Element => {
   const { t } = useAnalyticsTranslation();
 
   const [page, setPage] = useState(1);
+  const [items, setItems] = useState<DiscriminatedItem[]>([]);
   const [searchQuery, setSearchQuery] = useState('');
-  const { data: accessibleItems, isLoading } = hooks.useAccessibleItems(
+  const {
+    data: accessibleItems,
+    isLoading,
+    error,
+  } = hooks.useAccessibleItems(
     { name: searchQuery },
     // get items cumulative
-    { pageSize: page * ITEM_PAGE_SIZE },
+    { pageSize: ITEM_PAGE_SIZE, page },
   );
 
-  if (accessibleItems?.data) {
+  useEffect(() => {
+    if (accessibleItems?.data) {
+      if (page === 1) {
+        setItems(accessibleItems?.data);
+      } else {
+        setItems((prev) => {
+          const newItems = accessibleItems.data.filter(
+            (item) => !prev.some((p) => p.id === item.id),
+          );
```

Comment:
I kind of see why you would want to prevent inserting duplicated items. Can you add a comment that explains why this is needed ?